jc/parsers/srt.py

```diff
@@ -0,0 +1,204 @@
+                "minutes": 2,
+                "seconds": 21,
+                "milliseconds": 609,
+                "timestamp": "00:02:21,609"
+            },
+            "text": "Very good, Lieutenant.\n"
+        },
+        ...
+    ]
+"""
+
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = 'SRT file parser'
+    author = 'Mark Rotner'
+    author_email = 'rotner.mr@gmail.com'
+    compatible = ['linux', 'darwin', 'cygwin', 'win32', 'aix', 'freebsd']
+    tags = ['standard', 'file', 'string']
+
+
+__version__ = info.version
+
+
+def _process(proc_data):
+    """
+    Final processing to conform to the schema.
+
+    Parameters:
+
+        proc_data:  (Dictionary) raw structured data to process
+
+    Returns:
+
+        List of Dictionaries representing an SRT document.
+    """
+
+    int_list = {'index'}
+    timestamp_list = {"start", "end"}
+    timestamp_int_list = {"hours", "minutes", "seconds", "milliseconds"}
+
+    for entry in proc_data:
+        # Converting {"index"} to int.
+        for key in entry:
+            if key in int_list:
+                entry[key] = jc.utils.convert_to_int(entry[key])
+
+            # Converting {"hours", "minutes", "seconds", "milliseconds"} to int.
+            if key in timestamp_list:
+                timestamp = entry[key]
+                for timestamp_key in timestamp:
+                    if timestamp_key in timestamp_int_list:
+                        timestamp[timestamp_key] = jc.utils.convert_to_int(timestamp[timestamp_key])
+
+    return proc_data
+
+
+def parse_timestamp(timestamp: str):
+    """
+    timestamp: "hours:minutes:seconds,milliseconds" --->
+    {
+        "hours": "hours",
+        "minutes": "minutes",
+        "seconds": "seconds",
+        "milliseconds": "milliseconds",
+        "timestamp": "hours:minutes:seconds,milliseconds"
+    }
+    """
+
+    hours, minutes, seconds_milliseconds = timestamp.split(':')
+    seconds, milliseconds = seconds_milliseconds.split(',')
+    return {
+        "hours": hours,
+        "minutes": minutes,
+        "seconds": seconds,
+        "milliseconds": milliseconds,
+        "timestamp": timestamp
+    }
+
+
+def parse(data: str, raw: bool = False, quiet: bool = False):
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        Dictionary. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output = []
+    if not jc.utils.has_data(data):
+        return raw_output
+
+    subtitle = {}
+    lines = list(filter(None, data.splitlines()))
+    for line in lines:
+        line = line.strip()
+        if line.isdigit():  # Reached a new subtitle.

```

Comment:
Clever - I like it but wonder if we need to tighten this one a bit more. Maybe accounting for a newline-only before the digit (except maybe the first one)? Just in case there is a subtitle with only a single number. Just a thought.